noir/noir-repo/test_programs/noir_test_success/should_fail_with_matches/src/main.nr

```diff
@@ -17,3 +17,65 @@ fn test_should_fail_with_runtime_match() {
 fn test_should_fail_without_runtime_match() {
     assert_eq(dep::std::hash::pedersen_commitment([27]).x, 0);
 }
+
+#[test(should_fail_with = "PedersenPoint { x: 0x1cea3a116d01eb94d568ef04c3dfbc39f96f015ed801ab8958e360d406503ce0, y: 0x2721b237df87234acc36a238b8f231a3d31d18fe3845fff4cc59f0bd873818f8 }")]
+fn test_should_fail_with_struct(){
+    let hash = dep::std::hash::pedersen_commitment([27]);
+    assert_eq(hash.x, 0, hash);
```

Comment:
```suggestion
    assert_eq(hash.x, 0, f"hash");
```

This just looks wrong in my mind and we should tighten up the types accepted here.

Reply:
The idea is to support failing with data. In tests you are matching against a string, but in JS you can parse the data and get a meaningful error payload that you can run logic against, with the set of possible errors described in the ABI. And for aztec, this will allow in public functions to be able to call a function and handle the error it reverted with.